enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/messaging/marshalling/CoreReplicatedContentMarshal.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2002-2018 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j Enterprise Edition. The included source
+ * code can be redistributed and/or modified under the terms of the
+ * GNU AFFERO GENERAL PUBLIC LICENSE Version 3
+ * (http://www.fsf.org/licensing/licenses/agpl-3.0.html) with the
+ * Commons Clause, as found in the associated LICENSE.txt file.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * Neo4j object code can be licensed independently from the source
+ * under separate terms from the AGPL. Inquiries can be directed to:
+ * licensing@neo4j.com
+ *
+ * More information is also available at:
+ * https://neo4j.com/licensing/
+ */
+package org.neo4j.causalclustering.messaging.marshalling;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedList;
+
+import org.neo4j.causalclustering.core.consensus.NewLeaderBarrier;
+import org.neo4j.causalclustering.core.consensus.membership.MemberIdSet;
+import org.neo4j.causalclustering.core.consensus.membership.MemberIdSetSerializer;
+import org.neo4j.causalclustering.core.replication.DistributedOperation;
+import org.neo4j.causalclustering.core.replication.ReplicatedContent;
+import org.neo4j.causalclustering.core.state.machines.dummy.DummyRequest;
+import org.neo4j.causalclustering.core.state.machines.id.ReplicatedIdAllocationRequest;
+import org.neo4j.causalclustering.core.state.machines.id.ReplicatedIdAllocationRequestSerializer;
+import org.neo4j.causalclustering.core.state.machines.locks.ReplicatedLockTokenRequest;
+import org.neo4j.causalclustering.core.state.machines.locks.ReplicatedLockTokenSerializer;
+import org.neo4j.causalclustering.core.state.machines.token.ReplicatedTokenRequest;
+import org.neo4j.causalclustering.core.state.machines.token.ReplicatedTokenRequestSerializer;
+import org.neo4j.causalclustering.core.state.machines.tx.ReplicatedTransaction;
+import org.neo4j.causalclustering.core.state.machines.tx.ReplicatedTransactionSerializer;
+import org.neo4j.causalclustering.core.state.storage.SafeChannelMarshal;
+import org.neo4j.causalclustering.messaging.EndOfStreamException;
+import org.neo4j.storageengine.api.ReadableChannel;
+import org.neo4j.storageengine.api.WritableChannel;
+
+import static java.util.Collections.singleton;
+import static org.neo4j.causalclustering.messaging.marshalling.ByteBufAwareMarshal.simple;
+
+public class CoreReplicatedContentMarshal extends SafeChannelMarshal<ReplicatedContent>
```

Comment:
Are these changes compatible with Raft V1?

Reply:
Yes. We need to verify though through an upgrade test or similar. I did test though before removing the old class.